### PR TITLE
Added support for Windows platform

### PR DESCRIPTION
**Pull request by mrkskwsnck**
The compiler used was GCC from the MinGW64 toolchain. The method used to switch endianness is defined in ia32intrin.h but called through x86intrin.h headers.

### CODE REVIEW

File: luapgsql.c [hunk 1]
```diff
@@ -32,7 +32,11 @@
 #define htobe64(x) OSSwapHostToBigInt64(x)
 #elif __linux__
 #include <endian.h>
+#elif _WIN32
+#include <x86intrin.h>
```
Comment (daurnimator) [inline luapgsql.c]: This isn't a standard header on windows

File: luapgsql.c [hunk 2]
```diff
@@ -32,7 +32,11 @@
...
+#define htobe64(x) _bswap64(x)
```
Comment (daurnimator) [inline luapgsql.c]: Use `htonll` instead. https://msdn.microsoft.com/en-us/library/windows/desktop/jj710199(v=vs.85).aspx
Or check `#if BYTE_ORDER == LITTLE_ENDIAN` and use `__builtin_bswap64`

**Comment by MrksKwsnck** [comment]
@daurnimator The function `ntoll` is not defined with MinGW64 and GCC, so I implemented the latter alternative.

File: luapgsql.c [hunk 3]
```diff
@@ -32,7 +32,16 @@
 #define htobe64(x) OSSwapHostToBigInt64(x)
 #elif __linux__
 #include <endian.h>
+#elif _WIN32
+#include <sys/param.h>
+#if BYTE_ORDER == LITTLE_ENDIAN
+#include <x86intrin.h>
```
Comment (daurnimator) [inline luapgsql.c]: I think `__builtin_bswap64` is a compiler builtin. You shouldn't need this include?

Comment (MrksKwsnck) [reply luapgsql.c]: You are right. I removed it.

File: luapgsql.c [hunk 4]
```diff
@@ -32,7 +32,15 @@
 #define htobe64(x) OSSwapHostToBigInt64(x)
 #elif __linux__
 #include <endian.h>
+#elif _WIN32
+#include <sys/param.h>
```
Comment (daurnimator) [inline luapgsql.c]: This doesn't seem to be a standard windows header either :(

Googling around it seems that including winnt.h and then using `#if REG_DWORD == REG_DWORD_LITTLE_ENDIAN` is the correct thing?

Comment (MrksKwsnck) [reply luapgsql.c]: Okay, it still works. I could not include winnt.h directly. Instead windows.h includes windef.h and this includes winnt.h, finally.

In fact I am not that much into C programming. So thanks for your advices.

**Comment by daurnimator** [comment]
@MrksKwsnck can you confirm the current code works with both mingw and msvc?

**Comment by daurnimator** [comment]
Otherwise, could you remove your redundant commits from this PR?

**Comment by MrksKwsnck** [comment]
Unfortunately, it does not compile using MSVC :-(

```
C:\luapgsql>cl /nologo /MD /O2 -c -Foluapgsql.obj -IC:/opt/lua/include luapgsql.c -IC:\opt\pgsql\include
luapgsql.c
luapgsql.c(194): warning C4090: "Funktion": Unterschiedliche "const"-Qualifizierer

C:\luapgsql>link -dll -def:pgsql.def -out:pgsql.dll C:/opt/lua/lua53.dll luapgsql.obj -libpath:C:\opt\pgsql\lib pq.lib
Microsoft (R) Incremental Linker Version 14.11.25547.0
Copyright (C) Microsoft Corporation.  All rights reserved.

C:/opt/lua/lua53.dll : fatal error LNK1107: Ungültige oder beschädigte Datei: Lesen bei 0x430 nicht möglich.
```

I tried compiling using *Microsoft Visual Stuidio Build Tools 2017*. As I am not experienced in working with MSVC, I need to investigate the issue.

**Comment by MrksKwsnck** [comment]
Some time has passed and what I have managed so far is to successfully compile using MSVC (Microsoft Visual Studio Build Tools 2017). However, I still was not able to test the resulting DLL file properly, because it lacks some more dependencies (also the Dependency Walker was not helpful here).

**Comment by mbalmer** [comment]
Sorry for not acting on this earlier, but it looks  your diff does not apply anymore.  Please re-submit agains current sources if windows support is still wanted.

**Pull request closed without merging**